libs/sdk-core/src/breez_services.rs

```diff
@@ -1368,6 +1368,38 @@ impl BreezServices {
             },
         })
     }
+
+    pub async fn create_offer(&self, req: CreateOfferRequest) -> Result<String> {
+        self.start_node().await?;
+        Ok(
+            self
+                .node_api
+                .create_offer(req)
+                .await?
+        )
+    }
+
+    pub async fn pay_offer(&self, req: PayOfferRequest) -> Result<SendPaymentResponse, SendPaymentError> {
+        self.start_node().await?;
+
+        let fetch_invoice_response = self.node_api
+            .fetch_invoice(FetchInvoiceRequest {
+                offer: req.offer.clone(),
+                amount_msat: req.amount_msat,
+                timeout: req.timeout,
+                payer_note: req.payer_note.to_owned()
+            })
+            .await?;
+
+        if let Some(new_amount) = fetch_invoice_response.new_amount_msat {
+            return Err(SendPaymentError::OfferAmountChanged { new_amount });
+        }
+
+        Ok(self.send_payment(SendPaymentRequest {
+            invoice: fetch_invoice_response.bolt12,
+            amount_msat: req.amount_msat
```

Comment:
Does explicitly setting the amount work here? I think the invoice may have the amount filled already and the call to send_payment may fail?

Reply:
Technically this is a redundant check of the changes. If the target CLN node is working properly, we expect it to return an invoice with the same amount_msat as we specified, thus the call to `send_payment` should not fail. It is not necessary to pass it to the method, so might remove it altogether.

Reply:
For bolt11 invoices this fails I think, because cln will reject the call to `pay`. Only for zero amount invoices are you allowed to set the amount.

Reply:
Sorry, what I meant was just strictly in regards to the bolt12 flow. I fixed the issue by not passing `amount_msat` at all, as we expect the fetched invoice to have the same amount as our `FetchInvoiceRequest` anyway.